src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProvider.java

```diff
@@ -93,7 +93,7 @@ public static StreamingClientProvider getStreamingClientProviderInstance() {
    * call close client manually as eviction is executed lazily
    */
   private StreamingClientProvider() {
-    this.streamingClientHandler = new StreamingClientHandler();
+    this.streamingClientHandler = new DirectStreamingClientHandler();
```

Comment:
We also need to be able to inject a test double here.
Did you plan for this to be done as a part of SNOW-1055561?

Reply:
We can inject it using `getStreamingClientProviderForTests`, which is used in the new and the previous tests.